src/App.tsx

```diff
@@ -1,19 +1,78 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from 'react';
+import { debounce } from 'lodash';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [hasClick, setHasClick] = useState(false);
+  const [person, setPerson] = useState<Person | null>(null);
+  const [focus, setFocus] = useState(false);
+  const [query, setQuery] = useState('');
+  const [aplliedQuery, setAplliedQuery] = useState('');
+
+  const queryDetention = useCallback(
+    debounce((newQuery: string) => setAplliedQuery(newQuery), 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    if (!aplliedQuery) {
+      return peopleFromServer;
+    }
+
+    const normalizedQuery = aplliedQuery.toUpperCase();
+
+    return peopleFromServer.filter(currentPerson => {
+      const name = currentPerson.name.toUpperCase();
+
+      return name.includes(normalizedQuery);
+    });
+  }, [aplliedQuery]);
+
+  useEffect(() => setFocus(false), [person, hasClick]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuery = event.currentTarget.value;
+
+    setQuery(newQuery);
+    queryDetention(newQuery);
+  };
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    currentPerson: Person,
+  ) => {
+    event.preventDefault();
+    setQuery(currentPerson.name);
+    setHasClick(!hasClick);
+    setPerson(currentPerson);
+    setFocus(false);
+  };
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {visiblePeople.length !== 0 && person && query ? (
+        <h1 className="title">
+          {`${person.name} (${person.born} - ${person.died})`}
+        </h1>
+      ) : (
+        <h1 className="title">No matching suggestions</h1>
+      )}
 
-      <div className="dropdown is-active">
+      <div className={`dropdown ${focus && 'is-active'}`}>
```

Comment:
Not fixed
![2023-08-16_14h37_09](https://github.com/mate-academy/react_autocomplete/assets/61904995/5215500d-00c3-4cf2-b471-f983c00bad5f)